### PR TITLE
tree-sitter 0.16.5 (new formula)

### PR DESCRIPTION
**Pull request by mattmassicotte**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Review by chenrui333 — APPROVED** [review]
lgtm

File: Formula/tree-sitter.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  sha256 "3c6357d1a153c43f481d07f1a8743ed8b3134a2934689d564795a0d083ff7c7e"
+  head "https://github.com/tree-sitter/tree-sitter.git"
+
+  def install
+    # this shell script invokes clang or CC, and builds a .a with ar
+    system "script/build-lib"
+
+    lib.install "libtree-sitter.a"
+    include.install "lib/include/tree_sitter"
+  end
```
Comment (SMillerDev) [inline Formula/tree-sitter.rb]: Shouldn't it just depend on rust and be installed with cargo?

Comment (mattmassicotte) [reply Formula/tree-sitter.rb]: This library is C, but does come with Rust bindings. It seems likely that building the bindings also build the C library and maybe even installs it in the same way. This formula just follows the published tree-sitter documentation for a standard C build/install.

Do you think I need to look into building all the bindings too? There are a few different languages supported, but I only have experience with the default C library.

Comment (SMillerDev) [reply Formula/tree-sitter.rb]: Apparently there's also a rust CLI. It might be good to include that one.

Comment (chenrui333) [reply Formula/tree-sitter.rb]: Yeah, looks like there is a Rust CLI part of the build process, any idea if we want to include that in here too?

Comment (mattmassicotte) [reply Formula/tree-sitter.rb]: I haven't ever used it before as part of my work with tree-sitter. But, I'll investigate and report back.

**Comment by mattmassicotte** [comment]
I've looked into the rust-based CLI a little. It's a bit out of my comfort zone, and I'm not sure that I want to invest too much more time into it. Thanks all, but I think I'll just close this.

**Pull request closed without merging**